### ISSUE TITLE
[compiler bug] Python MLIR doesn't catch incorrect arguments to rotation gates

### ISSUE THREAD

**Issue by anthony-santana** [NVIDIA/cuda-quantum#1641]
### Required prerequisites

- [X] Consult the [security policy](https://github.com/NVIDIA/cuda-quantum/security/policy). If reporting a security vulnerability, do not report the bug using this form. Use the process described in the policy to report the issue.
- [X] Make sure you've read the [documentation](https://nvidia.github.io/cuda-quantum/latest). Your issue may be addressed there.
- [X] Search the [issue tracker](https://github.com/NVIDIA/cuda-quantum/issues) to verify that this hasn't already been reported. +1 or comment there if it has.
- [ ] If possible, make a PR with a failing test to give us a starting point to work on!

### Describe the bug

In python MLIR, we aren't catching when a user calls rotation gates with incorrect arguments.

### Steps to reproduce the bug

```
@cudaq.kernel
def kernel():
    q = cudaq.qubit()
    # Examples of things we can get away with without error:
    rx(3.14)
    rx("random_argument")
    # Our regular gates do throw an error like this:
    # x("random_argument") 
   

print(kernel)

result = cudaq.sample(kernel)
print(result)
```

### Expected behavior

Expect an error similar to the conventional gates, such as:

```
cudaq.kernel.ast_bridge.CompilerError: test.py:37: error: quantum operation x on incorrect quantum type <TYPENAME>.
         (offending source -> <VALUE>)
```

### Is this a regression? If it is, put the last known working version (or commit) here.

Not a regression

### Environment

- **CUDA Quantum version**: 
- **Python version**: 
- **C++ compiler**: 
- **Operating system**: 
All latest

**Comment by anthony-santana** [comment]
I do also think the general error message could use some improvement. Something that mentions function arguments. E.g

```
cudaq.kernel.ast_bridge.CompilerError: test.py:37: error: quantum operation <OP_NAME> received incorrect argument type <TYPENAME>.
         (offending source -> <VALUE>)
```

**Comment by amccaskey** [comment]
@anthony-santana you could add the following 
```diff
diff --git a/python/cudaq/kernel/ast_bridge.py b/python/cudaq/kernel/ast_bridge.py
index 412174880..649e5618e 100644
--- a/python/cudaq/kernel/ast_bridge.py
+++ b/python/cudaq/kernel/ast_bridge.py
@@ -1365,11 +1365,16 @@ class PyASTBridge(ast.NodeVisitor):
 
             if node.func.id in ["rx", "ry", "rz", "r1"]:
                 numValues = len(self.valueStack)
+                if numValues < 2:
+                    self.emitFatalError(f'Invalid number of arguments ({numValues}) passed to {node.func.id} (requires at least 2 arguments)', node)
                 qubitTargets = [self.popValue() for _ in range(numValues - 1)]
                 qubitTargets.reverse()
                 param = self.popValue()
                 if IntegerType.isinstance(param.type):
                     param = arith.SIToFPOp(self.getFloatType(), param).result
+                if not F64Type.isinstance(param.type):
+                    self.emitFatalError('rotational parameter must be a float.', node)
+                self.checkControlAndTargetTypes([], qubitTargets)
                 self.__applyQuantumOperation(node.func.id, [param],
                                              qubitTargets)
                 return
```

**Comment by khalatepradnya** [comment]
> Wondering if this fix would also cover the following issue #670

It's not the same. I think it was fixed by something else since the issue was filed. I will add that as a test as well.
